packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,267 @@
+    context.font = theme.font;
+
+    const fontWidth =
+      fontWidths.get(context.font) ?? GridRenderer.DEFAULT_FONT_WIDTH;
+    const truncationChar = model.truncationCharForCell(modelColumn, modelRow);
+    const truncatedText = GridRenderer.truncateToWidth(
+      context,
+      text,
+      textWidth,
+      fontWidth,
+      truncationChar
+    );
+
+    let lengthOfContent = text.indexOf(' ', truncatedText.length);
+    if (lengthOfContent === -1) {
+      lengthOfContent = Math.min(5000, text.length);
+    } else if (lengthOfContent > 5000) {
+      lengthOfContent = 5000;
+    }
+
+    const contentToCheckForLinks = text.substring(0, lengthOfContent);
+
+    const links = model.findLinksInText(contentToCheckForLinks);
+
+    // Check if the truncated text contains a link
+    if (links.length > 0) {
+      const {
+        actualBoundingBoxAscent,
+        actualBoundingBoxDescent,
+      } = context.measureText(truncatedText);
+      const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+
+      // Consider edge cases
+      const top = Math.max(gridY, gridY + textY - linkHeight / 2);
+      const bottom = clamp(
+        top + linkHeight,
+        top,
+        gridHeight - horizontalBarHeight
+      );
+
+      // Loop through the blocks of text
+      for (let i = 0; i < links.length; i += 1) {
+        const { end, start, href } = links[i];
+        let { value } = links[i];
+
+        if (end > truncatedText.length) {
+          value = truncatedText.substring(start);
+        }
+
+        let { width: linkWidth } = context.measureText(value);
+        // Measure the width of the substring before the link
+        const startX =
+          context.measureText(truncatedText.substring(0, start)).width + textX;
+        // Right side is greater than gridX
+        const textIsVisible = startX + linkWidth >= gridX;
+
+        if (textIsVisible) {
+          // Check if the x position is less than the grid x, then linkWidth should be shifted by gridX - startX
+          linkWidth -= startX < gridX ? gridX - startX : 0;
+
+          const left = Math.max(gridX, gridX + startX);
+          const right = clamp(
+            left + linkWidth,
+            left,
+            gridWidth - verticalBarWidth
+          );
+
+          const cursorIsInBounds =
+            x >= left && x <= right && y >= top && y <= bottom;
+
+          if (cursorIsInBounds) {
+            // Reset font
+            context.restore();
+
+            this.currentLinkBox = {
+              left,
+              top,
+              right,
+              bottom,
+              width: linkWidth,
+              height: linkHeight,
+              value,
+              href,
+            };
+
+            return this.currentLinkBox;
+          }
+        }
+      }
+    }
+
+    // There are no links or the mouse is not on a link, restore context and currentLinkBox
+    context.restore();
+    this.currentLinkBox = undefined;
+
+    return null;
```

Comment:
Do this check and early return before the if statement above. Then you can just continue the bulk of the logic without it being nested inside an if. This is a general thing I'd recommend when writing code since it helps reduce nesting

Instead of
```typescript
if (whatWeWant) {
  // Do things
  // Do more things
  // Do more things
  // cleanup and return
}

// Cleanup and return what we don't want
```

Do this so the bulk of the code is less indented
```typescript
if (whatWeDontWant) {
  // Cleanup and return what we don't want
}

// Do things
// Do more things
// Do more things
// cleanup and return
```